test/expect-tests/dune_rpc_e2e/dune_rpc_diagnostics.ml

```diff
@@ -814,6 +814,93 @@ let g = A.f
           ; [ "severity"; "error" ]
           ; [ "targets"; [] ]
           ]
-        ] |}]);
-  [%expect {||}]
+        ] |}])
+;;
+
+let%expect_test "cyclic dependency error simple" =
+  setup_diagnostics (fun client ->
+    files
+      [ "dune", "(executable (name foo))"; "foo.ml", "open Bar"; "bar.ml", "open Foo" ];
+    let* poll = poll_exn client Dune_rpc.Public.Sub.diagnostic in
+    let* () = print_diagnostics poll in
+    [%expect
+      {|
+        <no diagnostics> |}];
+    let* () = dune_build client "./foo.exe" in
+    [%expect
+      {|
+        Building ./foo.exe
+        Build ./foo.exe failed |}];
+    let+ () = print_diagnostics poll in
+    [%expect
+      {|
+      [ "Add"
+      ; [ [ "id"; "0" ]
+        ; [ "message"
+          ; [ "Verbatim"
+            ; "Error: dependency cycle between modules in _build/default:\n\
+              \   Foo\n\
+               -> Bar\n\
+               -> Foo\n\
+               "
+            ]
+          ]
+        ; [ "promotion"; [] ]
+        ; [ "related"; [] ]
+        ; [ "severity"; "error" ]
+        ; [ "targets"; [] ]
+        ]
+      ]
+      |}])
+;;
+
+let%expect_test "cyclic dependency error" =
+  setup_diagnostics (fun client ->
+    files
+      [ "dune", "(executable (name foo))"
+      ; "foo.ml", "open Bar open Baz"
+      ; "bar.ml", "open Baz"
+      ; "baz.ml", "open Bar"
+      ];
+    let* poll = poll_exn client Dune_rpc.Public.Sub.diagnostic in
+    let* () = print_diagnostics poll in
+    [%expect
+      {|
+        <no diagnostics> |}];
+    let* () = dune_build client "./foo.exe" in
+    [%expect
+      {|
+        Building ./foo.exe
+        Build ./foo.exe failed |}];
+    let+ () = print_diagnostics poll in
+    [%expect
+      {|
+      [ "Add"
+      ; [ [ "id"; "0" ]
+        ; [ "message"
+          ; [ "Verbatim"
+            ; "Cycle_error.E\n\
+              \  [ (\"Rule.make\", ())\n\
```

Comment:
Could you filter out this stack trace from this output?

Reply:
Done.